src/components/source_editor/LatticeConfigurationDialog.js

```diff
@@ -108,6 +109,10 @@ class LatticeConfigurationDialog extends React.Component {
         const newMaterial = new Material(newMaterialConfig);
         // assert basis is stored in 'crystal' units
         newMaterial.toCrystal();
+
+        // update to non-periodic if asked to do so
+        this.state.isNonPeriodic ? newMaterial.toCartesian() : newMaterial.toCrystal();
+        newMaterial.updateIsNonPeriodic(this.state.isNonPeriodic);
```

Comment:
Why is an update to is non-periodic handled inside the lattice update???

The two are independent, we discussed having the switch at the top of the lattice dialogue and should isolate it.

When the non-periodic switch is triggered, the rest of the lattice form fields should be inactive